src/main/scala/wdlTools/generators/code/package.scala

```diff
@@ -44,6 +44,94 @@ trait Sized {
   def firstLineLength: Int = length
 }
 
+trait ExpressionState {
+  def canAdvanceTo(state: ExpressionState): Boolean
+}
+
+trait InitialState extends ExpressionState {
+  override def canAdvanceTo(state: ExpressionState): Boolean = {
+    state match {
+      case _: InStringState | _: InOperationState => true
+      case _                                      => false
+    }
+  }
+}
+
+/**
+  * The initial expression state.
+  */
+object StartState extends InitialState
+
+/**
+  * The state of being within a string expression but not within a placeholder.
+  * @param quoting the type of quoting for the string
+  */
+case class InStringState(quoting: Quoting.Quoting) extends ExpressionState {
+  override def canAdvanceTo(state: ExpressionState): Boolean = {
+    state match {
+      case _: InStringState   => true
+      case InPlaceholderState => true
```

Comment:
I see that the only way to reach InPlaceholderState is through InStringState. I think I have in the past used ~{expr} directly in the command block without enclosing it in "", but according to this is that non-standard WDL, and those expressions should always be in a string context? https://github.com/openwdl/wdl/blob/main/versions/1.1/SPEC.md#expression-placeholders-and-string-interpolation

Reply:
The entire command block is implicitly a string without enclosing quotes. https://github.com/openwdl/wdl/blob/main/versions/1.1/SPEC.md#expression-placeholders